### PR TITLE
Add auto-generated product layout

### LINKED ISSUE #122: Add auto-rendered product layout
Add auto-rendered product layout where the product data is rendered based on [product drop](https://www.voog.com/developers/markup/ecommerce-objects/product) data.

It should look like current "Product" layout. Use new [image](https://www.voog.com/developers/markup/tags/image) or [image_data](https://www.voog.com/developers/markup/tags/image-data) tag for product cover image.

It content type must be `product`:

```json
{
  "content_type": "product",
  "component": false,
  "file": "layouts/auto_generated_product.tpl",
  "layout_name": "auto_generated_product",
  "title": "Auto-generated product"
}
```

See also buy button rendering example in [Nuuk design](https://github.com/Voog/design-nuuk/blob/master/components/buy-button.tpl).

## TODO

* [ ] Add "Auto-generated product" layout
* [ ] Check and improve "[product list](https://www.voog.com/support/online-store/product-list-content-area)" widget styles when needed.

### PR DESCRIPTION
**Pull request by pxska**
Adds an auto-generated product layout to the design based on the initial product layout. Also add a browser hack to render crispy images in Chrome.

Closes #122.

### CODE REVIEW

File: components/buy-button.tpl [hunk 1]
```diff
@@ -0,0 +1,55 @@
+{%- capture _button_attributes %}
+  data-product-id="{{ product.id }}"
+  data-product="{{ product | json | escape }}"
+  data-settings="{&quot;title&quot;:&quot;{{ "add_to_cart" | lc | escape_once }}&quot;,&quot;button_style&quot;:&quot;with_price&quot;}"
```
Comment (tanelj) [inline components/buy-button.tpl:4]: The price is allready on the page.

```suggestion
  data-settings="{&quot;title&quot;:&quot;{{ "add_to_cart" | lc | escape_once }}&quot;}"
```

File: layouts/auto_generated_product.tpl [hunk 2]
```diff
@@ -0,0 +1,118 @@
+<!DOCTYPE html>
+<html class="{% if editmode %}editmode{% else %}public{% endif %}" lang="{{ page.language_code }}">
+
+<head prefix="og: http://ogp.me/ns#">
+  {% include "template-variables" %}
+  {% include "html-head" %}
+  {% include "template-styles" %}
+</head>
+
+{% capture bottom_content_html %}
```
Comment (tanelj) [inline layouts/auto_generated_product.tpl:10]: Let's strip white spaces here and below (until body)

```suggestion
{%- capture bottom_content_html %}
```

File: sources/javascripts/concat/global/main.js [hunk 3]
```diff
@@ -118,6 +118,35 @@
...
+    });
+
+    $(window).resize(debounce(function () {
+      changeProductImagePos();
+    }, 25));
+
+    var changeProductImagePos = function () {
+      var productGallery = $('.js-product-gallery');
+      var productImageContentBox = $('.js-content-item-box');
+      var buyBtnContent = $('.js-buy-btn-content');
```
Comment (tanelj) [inline sources/javascripts/concat/global/main.js:134]: Mayby its more readable to add class like `js-product-content-right` to product right column wrapper and use it here (like in Santiago https://github.com/Voog/design-santiago/pull/230/files#diff-96d8063d0f25fce87494d9213aa996cf6bf3e4288934f4e77634d45ee6abfe82R1075)

**Review by tanelj — CHANGES_REQUESTED** [review]
Very good, but some minor comments!

Please also add
* [x] Webkit image fix https://github.com/Voog/design-santiago/commit/7d6f73c1424e946cf108c3f5bb3475488ac2bca9#diff-d543c7ef54a2ad318b0bd7a534b83515eab150ff158798967ce5d6953b8783ad

File: layouts/auto_generated_product.tpl [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+            <div class="flex_row-2--item-40 js-product-content-right">
+              <div class="mar_0-32 flex_col">
+                <section class="content-body content-formatted" data-search-indexing-allowed="true">
+                  <div class="content-item-title content-area">
+                    <h1>{%- editable product.name -%}</h1>
+                  </div>
+
+                  <div class="product-price">
+                    {%- if product.price_max_with_tax != product.price_min_with_tax -%}
+                    {{ product.price_min_with_tax | money_with_currency: product.currency -}}
```
Comment (PeeterPaal) [inline layouts/auto_generated_product.tpl:73]: Missing indent

**Review by PeeterPaal — CHANGES_REQUESTED** [review]
Seems nice! Just a few nitpicks:

The margins seem too large. This makes the content area under product image quite narrow (compared to regular `Product` layout).

**Review by tanelj — APPROVED** [review]
Nice!

**Review by PeeterPaal — APPROVED** [review]
Looks good!

File: components/template-variables.tpl [hunk 5]
```diff
@@ -1,4 +1,11 @@
 {% capture dont_render %}
+  {% comment %}Detects viewing mode.{% endcomment %}
```
Comment (tanelj) [inline components/template-variables.tpl:2]: Separate `view_mode` is not needed. We have all ready `html.editmode` class.

File: layouts/auto_generated_product.tpl [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+      </div>
+    </div>
+  </div>
+
+  {% include "site-signout" %}
+  {% include "javascripts" %}
+  {% include "template-tools" %}
+
+  <script>
+    template && template.handleProductPageContent();
```
Comment (tanelj) [inline layouts/auto_generated_product.tpl:117]: Let's use only `window.site`.

File: layouts/product.tpl [hunk 7]
```diff
@@ -67,6 +67,12 @@
               <div class="mar_0-32 flex_col">
                 <section class="content-body content-formatted js-buy-btn-content mar_32-0"
                   data-search-indexing-allowed="true">
+                  <div class="content-item-title content-area" data-search-indexing-allowed="true">
```
Comment (tanelj) [inline layouts/product.tpl:70]: `data-search-indexing-allowed="true"` is all ready presented in prev line.

```suggestion
                  <div class="content-item-title content-area">
```

File: sources/javascripts/concat/global/main.js [hunk 8]
```diff
@@ -176,6 +205,10 @@
     bindSiteSearch: bindSiteSearch
   });
 
+  window.template = $.extend(window.template || {}, {
```
Comment (tanelj) [inline sources/javascripts/concat/global/main.js:208]: We don't need separate `window.template`. Please use use `window.site`.

**Review by tanelj — CHANGES_REQUESTED** [review]
Works, but some technical remarks.

**Review by PeeterPaal — APPROVED** [review]
Looks good!

**Review by tanelj — APPROVED** [review]

**Pull request merged**